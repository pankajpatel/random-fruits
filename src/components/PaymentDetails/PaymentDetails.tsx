import styled from 'styled-components';
import { FormattedDate, FormattedMessage } from 'react-intl';

import { Badge } from '@app/ds/Badge';
import { Label } from '@app/ds/Label';
import { getBadgeColorForStatus } from '@app/utils/getBadgeColorForStatus';
import { Amount } from '../Amount/Amount';

const Container = styled.div`
  padding: 0.5rem;
  display: grid;
  grid-template-columns: 1fr 1fr;

  ${Label} + span {
    font-size: 2rem;
    display: block;
    margin-bottom: 1.5rem;
  }
`;

export const PaymentDetails = ({
  payment,
}: {
  payment: Payment;
}): JSX.Element => (
  <Container>
    <div>
      <Label>
        <FormattedMessage id="label.customer" defaultMessage="Customer" />
      </Label>
      <span>{payment.customer_name}</span>
    </div>
    <div>
      <Label>
        <FormattedMessage id="label.merchant" defaultMessage="Merchant" />
      </Label>
      <span>{payment.merchant.name}</span>
    </div>
    <div>
      <Label>
        <FormattedMessage id="label.amount" defaultMessage="Amount" />
      </Label>
      <span>
        <Amount value={payment.amount} />
      </span>
    </div>
    <div>
      <Label>
        <FormattedMessage id="label.created" defaultMessage="Initiated On" />
      </Label>
      <span>
        <FormattedDate value={new Date(payment.created)} dateStyle="full" />
      </span>
    </div>
    <div>
      <Label>
        <FormattedMessage id="label.status" defaultMessage="" />
      </Label>
      <span>
        <Badge type={getBadgeColorForStatus(payment.status)}>
          <FormattedMessage
            id={`status.${payment.status}`}
            defaultMessage={payment.status}
          />
        </Badge>
      </span>
    </div>
  </Container>
);
