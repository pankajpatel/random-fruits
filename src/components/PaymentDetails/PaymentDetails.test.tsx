import { render, screen } from '@testing-library/react';

import { withProviders } from '@app/testing/provider-setup';
import mockPaymentJSON from '@app/fixtures/payment.json';

import { PaymentDetails } from './PaymentDetails';

describe('Payment Details', () => {
  it('should render without crashing', () => {
    const { container } = render(
      withProviders(<PaymentDetails payment={mockPaymentJSON as Payment} />)
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with payment data', () => {
    render(
      withProviders(<PaymentDetails payment={mockPaymentJSON as Payment} />)
    );
    expect(screen.getByText('label.customer')).toBeInTheDocument();
    expect(screen.getByText(mockPaymentJSON.customer_name)).toBeInTheDocument();
    expect(screen.getByText('label.merchant')).toBeInTheDocument();
    expect(screen.getByText(mockPaymentJSON.merchant.name)).toBeInTheDocument();
    expect(screen.getByText('label.amount')).toBeInTheDocument();
    expect(screen.getByText(`${mockPaymentJSON.amount} â‚¬`)).toBeInTheDocument();
    expect(screen.getByText('label.created')).toBeInTheDocument();
    expect(screen.getByText('label.status')).toBeInTheDocument();
    expect(
      screen.getByText(`status.${mockPaymentJSON.status}`)
    ).toBeInTheDocument();
  });
});
