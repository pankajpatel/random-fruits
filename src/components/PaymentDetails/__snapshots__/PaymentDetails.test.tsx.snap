// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Details should render without crashing 1`] = `
.c3 {
  padding: 0.35em 0.55em;
  border-radius: 0.4em;
  background-color: #e3f2fd;
  border: 1px solid rgba(0,0,0,0.05);
  line-height: 1;
}

.c2 {
  display: block;
  font-size: 1.2rem;
  color: #aaa;
}

.c0 {
  padding: 0.5rem;
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.c0 .c1 + span {
  font-size: 2rem;
  display: block;
  margin-bottom: 1.5rem;
}

<div>
  <div
    class="c0"
  >
    <div>
      <label
        class="c1 c2"
      >
        label.customer
      </label>
      <span>
        Émile Zola
      </span>
    </div>
    <div>
      <label
        class="c1 c2"
      >
        label.merchant
      </label>
      <span>
        Super Merchant
      </span>
    </div>
    <div>
      <label
        class="c1 c2"
      >
        label.amount
      </label>
      <span>
        356050
         €
      </span>
    </div>
    <div>
      <label
        class="c1 c2"
      >
        label.created
      </label>
      <span>
        Mon Apr 04 2022 00:00:00 GMT+0200 (Central European Summer Time)
      </span>
    </div>
    <div>
      <label
        class="c1 c2"
      >
        label.status
      </label>
      <span>
        <span
          class="c3"
          type="info"
        >
          status.ready
        </span>
      </span>
    </div>
  </div>
</div>
`;
